File: src/main/java/view/utils/ComponentsBuilder.java
package view.utils;

import javax.swing.*;
import java.util.Vector;

public class ComponentsBuilder {

	public static JList<String> createList( Vector<String> items, int row, int cellW ) {
		JList<String> list = new JList<>( new String[] { "" } );
		list.setSelectionMode( ListSelectionModel.SINGLE_INTERVAL_SELECTION );
		list.setLayoutOrientation( JList.VERTICAL );
		list.setVisibleRowCount( row );
		list.setFixedCellHeight( items.size( ) < 10
				? list.getPreferredSize( ).height * 5
				: Math.min( items.size( ), 25 ) );
		list.setFixedCellWidth( cellW );
		list.setListData( items );
		return list;
	}
}

File: client/src/main/kotlin/view/util/ComponentsBuilder.kt
package view.util

import java.util.*
import javax.swing.JList
import javax.swing.ListSelectionModel

/**
 * An utility object which contains a method to comfortably create lists
 *
 * @author Paolo Baldini
 */
object ComponentsBuilder {
    fun createList(items: Vector<String>, row: Int, cellW: Int): JList<String> {
        val list = JList(arrayOf(""))
        list.selectionMode = ListSelectionModel.SINGLE_INTERVAL_SELECTION
        list.layoutOrientation = JList.VERTICAL
        list.visibleRowCount = row
        list.fixedCellHeight = if (items.size < 10) list.preferredSize.height * 5 else Math.min(items.size, 25)
        list.fixedCellWidth = cellW
        list.setListData(items)
        return list
    }
}